### ISSUE TITLE
"react/react-in-jsx-scope" should support jsx pragma definitions in different types of comments

### ISSUE THREAD

**Issue by atoiv** [yannickcr/eslint-plugin-react#1360]
It seems that when defining `@jsx` pragma  in a file, "react/react-in-jsx-scope" requires pragma to be defined in a specific comments style in order to work.

Rule works correctly if jsx pragma is defined with JSDoc style comments:

```js
/** @jsx foo */
```

Rule doesn't work when using regular multiline comment:

```js
/* @jsx foo */
```

The implementation of the rule seems to match Babel's [transform-react-jsx](https://babeljs.io/docs/plugins/transform-react-jsx) documentation, which shows examples only using JSDoc comments`/** ... */`. 

The transfom-react-jsx plugin also supports regular multiline comments `/* ... */`.

**Comment by ljharb** [comment]
jsdoc-style comments generally always require `/**` - the convention is that the extra star is an explicit signal. i don't think it'd be a good idea to cover single-star comments.

**Comment by atoiv** [comment]
True, except jsdoc style is commonly used for generating documentation,
where as jsx pragma explicitly affects code generation.

Besides, as the babel plugin supports regular comments it's not obvious
that this rule doesn't do the same.

**Comment by ljharb** [comment]
I've looked; I've not found a single example that uses `/*` - they all use `/**`.

If you can cite a source, i'd be happy to reopen this.
